File: Arrays/Algo_Insertion_sort.js
function insertionSort(arr)
{
    for(var i=1;i<arr.length;i++)
    {
        if(arr[i]<arr[i-1])
        {
            var temp=arr[i];
            for(var j=i-1;j>=0;j--)
            {
                arr[j+1]=arr[j];
                if(j==0 || temp>arr[j-1])
                {
                    arr[j]=temp;
                    break;
                }
            }
        }
    }
}

var arr = [6,5,4,3,0,1];
insertionSort(arr);
console.log(arr);

File: Arrays/Algo_Insertion_sort.py
def insertionSort(arr):
    for i in range(len(arr)):
        if (arr[i]<arr[i-1]):
            temp=arr[i]
            for j in range(i-1,-1,-1):
                arr[j+1]=arr[j]
                if(j==0 or temp>arr[j-1]):
                    arr[j]=temp
                    break

arr = [6,5,4,3,0,1]
insertionSort(arr)
print(arr)